Yoast/wordpress-seo
# Add Japanese folder and Researcher

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds Japanese folder and Researcher.

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Build the free plugin
* Enable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` in `basic-wordpress-config.php` file
* Set your site to Japanese` 日本語`
* Create/open a post
* In the console, enter `window.yoast.Researcher` and confirm that it returns as shown below (the Researcher file should start with `ja`):
<img width="537" alt="Screenshot 2021-09-09 at 12 15 22" src="https://user-images.githubusercontent.com/48715883/132668583-4066a162-49f7-415d-aafd-a2431865142b.png">


* Disable the `JAPANESE_SUPPORT` by adding this  `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` in `basic-wordpress-config.php` file
* Reload your page
* Confirm that you get the default Researcher by entering `window.yoast.Researcher` in the console:
<img width="538" alt="Screenshot 2021-09-09 at 12 17 44" src="https://user-images.githubusercontent.com/48715883/132668723-7e328f42-7333-46c0-98f0-09a435d4e5f5.png">


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* Install and activate Free
* Enable the `JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` in `basic-wordpress-config.php` file
* Set your site to Japanese` 日本語`
* Create/open a post
* In the console, enter `window.yoast.Researcher` and confirm that it returns as shown below (the Researcher file should start with `ja`):
<img width="537" alt="Screenshot 2021-09-09 at 12 15 22" src="https://user-images.githubusercontent.com/48715883/132668583-4066a162-49f7-415d-aafd-a2431865142b.png">


* Disable the `JAPANESE_SUPPORT` by adding this  `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` in `basic-wordpress-config.php` file
* Reload your page
* Confirm that you get the default Researcher by entering `window.yoast.Researcher` in the console:
<img width="538" alt="Screenshot 2021-09-09 at 12 17 44" src="https://user-images.githubusercontent.com/48715883/132668723-7e328f42-7333-46c0-98f0-09a435d4e5f5.png">

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-1022
